File: DemoClient/projects/ng-realtime-database/src/lib/models/command/query-command.ts
import {CollectionCommandBase} from './collection-command-base';
import {IPrefilter} from '../prefilter/iprefilter';

export class QueryCommand extends CollectionCommandBase {
  prefilters: IPrefilter<any>[];

  constructor(collectionName: string, prefilters: IPrefilter<any>[]) {
    super('QueryCommand', collectionName);
    this.prefilters = prefilters;
  }
}

File: RealtimeDatabase/Command/Query/QueryCommand.cs
using System.Collections.Generic;
using RealtimeDatabase.Internal.Prefilter;

namespace RealtimeDatabase.Command.Query
{
    public class QueryCommand : CollectionCommandBase
    {
        public List<IPrefilterBase> Prefilters { get; set; } = new List<IPrefilterBase>();
    }
}
